### PR TITLE
TW-1479: [EVM] Transactions history with Alchemy

### PR DESCRIPTION
**Pull request by alex-tsx**
https://madfish.atlassian.net/browse/TW-1479

### CODE REVIEW

File: src/routers/evm/alchemy.ts [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+function sortPredicate(
+  { metadata: { blockTimestamp: aTs } }: AssetTransfersWithMetadataResult,
+  { metadata: { blockTimestamp: bTs } }: AssetTransfersWithMetadataResult
+) {
+  if (aTs < bTs) return 1;
+  if (aTs > bTs) return -1;
+  // return aTs < bTs ? 1 : -1;
+
+  return 0;
```
Comment (keshan3262) [inline src/routers/evm/alchemy.ts:183]:
```suggestion
  return bTs.localeCompare(aTs);
```

File: src/routers/evm/alchemy.ts [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+  fromBlock?: string
+) {
+  return alchemy.core.getLogs({
+    address: contractAddress,
+    topics: [
+      [
+        '0x8c5be1e5ebec7d5bd14f71427d1e84f3dd0314c0f7b2291e5b200ac8c7c3b925', // Approval
+        '0x17307eab39ab6107e8899845ad3d59bd9653f200f220920489ca2b5937696c31' // ApprovalForAll
+      ],
+      `0x000000000000000000000000${accAddress.slice(2)}`
```
Comment (keshan3262) [inline src/routers/evm/alchemy.ts:201]:
```suggestion
      `0x${accAddress.slice(2).padStart(64, '0')}`
```
This option looks more informative to me

File: src/routers/evm/alchemy.ts (hunk 1 above)
Comment (keshan3262) [reply src/routers/evm/alchemy.ts]: I agree

File: src/routers/evm/alchemy.ts [hunk 3]
```diff
@@ -178,31 +178,38 @@ function sortPredicate(
...
-      `0x000000000000000000000000${accAddress.slice(2)}`
-    ],
-    toBlock: olderThanBlockToToBlockValue(olderThanBlockHeight),
-    fromBlock
-  });
+      toBlock,
+      fromBlock
+    });
+  } catch (error: any) {
+    // For 'query exceeds max block range 100000'
```
[the remark this answers is not in the document]
Comment (keshan3262) [reply src/routers/evm/alchemy.ts:208]: Okay but we can save even more compute units if we do not make requests that will definitely fail

[the remark this answers is not in the document]
Comment (keshan3262) [reply src/routers/evm/alchemy.ts:208]: This comment suggests that we do them when the block range is over 100000 although we know that the request will fail

**Review by keshan3262 — APPROVED** [review]

**Review by lendihop — APPROVED** [review]

**Pull request merged**